### PR TITLE
Reorganize FileIO benchmarks

### PR DESCRIPTION
**Pull request by harendra-kumar**
Better reviewed by individual commits.

### CODE REVIEW

File: benchmark/Streamly/Benchmark/FileSystem/Handle.hs [hunk 1]
```diff
@@ -0,0 +1,1351 @@
...
+-- copy stream
+-------------------------------------------------------------------------------
+
+-- | Send the file contents to /dev/null
+{-# INLINE readFromBytesNull #-}
+readFromBytesNull :: Handle -> Handle -> IO ()
+readFromBytesNull devNull inh = IFH.fromBytes devNull $ S.unfold FH.read inh
+
+#ifdef INSPECTION
+inspect $ hasNoTypeClasses 'catStreamWrite
```
Comment (adithyaov) [inline benchmark/Streamly/Benchmark/FileSystem/Handle.hs]: `catStreamWrite`? Maybe you changed the name.

File: benchmark/Streamly/Benchmark/FileSystem/Handle.hs [hunk 2]
```diff
@@ -0,0 +1,1351 @@
...
+
+{-# INLINE toChunksWithBufferOfDecodeUtf8ArraysLenient #-}
+toChunksWithBufferOfDecodeUtf8ArraysLenient :: Handle -> IO ()
+toChunksWithBufferOfDecodeUtf8ArraysLenient inh =
+   S.drain
+     $ IUS.decodeUtf8ArraysLenient
+     $ IFH.toChunksWithBufferOf (1024*1024) inh
+
+#ifdef INSPECTION
+inspect $ hasNoTypeClasses 'decodeUtf8Lenient
```
Comment (adithyaov) [inline benchmark/Streamly/Benchmark/FileSystem/Handle.hs]: `decodeUtf8Lenient` does not exist I here? Maybe you meant `toChunksWithBufferOfDecodeUtf8ArraysLenient`?

File: benchmark/Streamly/Benchmark/FileSystem/Handle.hs [hunk 3]
```diff
@@ -0,0 +1,1351 @@
...
+
+-- | Copy file
+{-# INLINE copyChunks #-}
+copyChunks :: Handle -> Handle -> IO ()
+copyChunks inh outh =
+    let s = IFH.toChunks inh
+    in S.fold (IFH.writeChunks outh) s
+
+#ifdef INSPECTION
+inspect $ hasNoTypeClasses 'toChunksWriteChunks
```
Comment (adithyaov) [inline benchmark/Streamly/Benchmark/FileSystem/Handle.hs]: `toChunksWriteChunks` does nor exist. Maybe you meant `copyChunks`?

File: benchmark/Streamly/Benchmark/FileSystem/Handle.hs [hunk 4]
```diff
@@ -0,0 +1,1351 @@
...
+toChunksConcatUnfoldCountLines :: Handle -> IO Int
+toChunksConcatUnfoldCountLines inh =
+    S.length
+        $ IUS.lines FL.drain
+        $ SS.decodeLatin1
+        -- XXX replace with toBytes
+        $ S.concatUnfold A.read (IFH.toChunks inh)
+
+#ifdef INSPECTION
+inspect $ hasNoTypeClasses 'countLinesU
```
Comment (adithyaov) [inline benchmark/Streamly/Benchmark/FileSystem/Handle.hs]: `countLinesU`?

**Comment by adithyaov** [comment]
I'm reviewing this commit-by-commit. Few comments might be addressed in a future commit in which case, please ignore them.

File: benchmark/Streamly/Benchmark/FileSystem/Handle.hs (hunk 2 above)
Comment (adithyaov) [reply benchmark/Streamly/Benchmark/FileSystem/Handle.hs]: This has been addressed in `bb6954f1a13553037e25862e3341f55e4e3ea0a2`.

File: benchmark/Streamly/Benchmark/FileSystem/Handle.hs (hunk 3 above)
Comment (adithyaov) [reply benchmark/Streamly/Benchmark/FileSystem/Handle.hs]: This has been addressed in `bb6954f1a13553037e25862e3341f55e4e3ea0a2`.

File: benchmark/Streamly/Benchmark/FileSystem/Handle.hs (hunk 1 above)
Comment (adithyaov) [reply benchmark/Streamly/Benchmark/FileSystem/Handle.hs]: This has been addressed in `bb6954f1a13553037e25862e3341f55e4e3ea0a2`.

File: benchmark/Streamly/Benchmark/FileSystem/Handle.hs (hunk 4 above)
Comment (adithyaov) [reply benchmark/Streamly/Benchmark/FileSystem/Handle.hs]: This has been addressed in `bb6954f1a13553037e25862e3341f55e4e3ea0a2`.

**Comment by adithyaov** [comment]
LGTM. Travis's failure seems to be due to some version conflict.

**Comment by harendra-kumar** [comment]
Thanks @adithyaov !

**Pull request merged**